app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.java

```diff
@@ -462,14 +490,40 @@ public void onCommit() {
                     ? UrlUtils.normalize(input)
                     : UrlUtils.createSearchUrl(getContext(), input);
 
+            if (session != null) {
+                if (!isUrl) {
+                    session.setSearch(true);
+                    session.setSearchTerms(input.trim());
+                } else {
+                    session.setSearch(false);
+                }
+            } else {
+                if (!isUrl) {
+                    isSearchNewSession = true;
+                    searchTermsNewSession = input.trim();
```

Comment:
Instead of keeping references in the UrlInputFragment instance we could pass those values to openUrl(), I guess?